vars/rapidastUtils.groovy

```diff
@@ -24,6 +24,35 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
         stage("Collect artifacts") {
             archiveArtifacts allowEmptyArchive: true, artifacts: "results/${ServiceName}/**/zap/*.*, , results.html"
         }
+
+        stage("Send data to Sitreps Grafana") {
+            // There is a a dir that contains a timestamp which would be harder to predict, instead try to find resource.
+            def RAPIDAST_JSON_REPORT = sh returnStdout: true, script: "find . -name zap-report.json"
+            String[] str = RAPIDAST_JSON_REPORT.split("/");
+            def random_results_dir = str[3]
+            def html = "${BUILD_URL}/artifact/results/${ServiceName}/${random_results_dir}/zap/zap-report.html"
```

Comment:
You can use stash and unstash to get json and html report here

https://linuxhint.com/stash-in-jenkins/

Reply:
we are not trying to send the report, but the url of the HTML report. So, we would need to know the path for viewable html url

Reply:
Ok, I think more simpler way here is to use https://plugins.jenkins.io/htmlpublisher/, which will give you the direct url of html file.

use `publishHTML()` in colelct artifacts step and you will get your url like `${BUILD_URL}/zap-report.html`, which will remove the workaround that you used to get url.